internal/autoevent/executor_test.go

```diff
@@ -13,21 +13,27 @@ import (
 )
 
 func TestCompareReadings(t *testing.T) {
-	readings := make([]contract.Reading, 3)
+	readings := make([]contract.Reading, 4)
 	readings[0] = contract.Reading{Name: "Temperature", Value: "10"}
 	readings[1] = contract.Reading{Name: "Humidity", Value: "50"}
 	readings[2] = contract.Reading{Name: "Pressure", Value: "3"}
+	readings[3] = contract.Reading{Name: "Image", BinaryValue: []byte("This is a image")}
```

Comment:
add at least one more positive case to test the scenario of unchanged binary value

Reply:
add the case of when hasBinary = false in compareReadings func (only simple readings)
hasBinary = true , but no binary data
hasBinary = false, but contain binary data